ledger/participant-integration-api/src/main/scala/platform/apiserver/ApiServer.scala

```diff
@@ -9,10 +9,13 @@ import scala.concurrent.Future
 
 trait ApiServer {
 
-  /** the API port the server is listening on */
+  /** The API port that the server is listening on. */
   def port: Port
 
-  /** completes when all services have been closed during the shutdown */
-  def servicesClosed(): Future[Unit]
+  /** A future that completes when the server is up and has a valid configuration. */
+  def ready: Future[Unit]
```

Comment:
Note that there are two kinds of "ready" in the ledger API server:
- At startup: All services that do not depend on the ledger configuration work immediately. Services that do depend on the ledger configuration (submission service and command service) may return `UNAVAILABLE`.
- After the ledger API server has seen at least one ledger configuration in the index: all services work.

The idea of the ledger API server is to wait at startup until it sees at least one ledger configuration (to avoid users accessing the server when it's not "fully" ready), but stop waiting if no configuration appears within a short time (in case there just is no configuration on the ledger and you need to set the initial configuration through the ledger API server). I'm not sure this method solves anything, it sounds like the test should just increase the timeout for waiting for the ledger config.

Details:
- `ApiServicesBundle.acquire()` [already waits](https://github.com/digital-asset/daml/blob/master/ledger/participant-integration-api/src/main/scala/platform/apiserver/ApiServices.scala#L113) on `ledgerConfigProvider.ready`
- `ledgerConfigProvider.ready` is resolved at the latest after [LedgerConfiguration.configurationLoadTimeout](https://github.com/digital-asset/daml/blob/master/ledger/participant-integration-api/src/main/scala/platform/configuration/LedgerConfiguration.scala#L16). If that timeout is set to a small value, then the ledger API server will appear to be "ready" even though the submission/command services are not.

Reply:
Facepalm. Of course it does; this doesn't solve the problem at all.

So here's the issue I was trying to solve, in `ScenarioLoadingITPostgres`:

```
2020-11-18T09:50:23.578Z INFO  c.d.p.a.s.LedgerConfigProvider@[kka.actor.default-dispatcher-9] -
                No ledger configuration found, submitting an initial configuration b81c7775-771d-4413-9288-bfcb6840f041 (context: {participantId=sandbox-participant})
2020-11-18T09:50:23.582Z INFO  c.d.p.a.s.LedgerConfigProvider@[ForkJoinPool-20-worker-9] -
                Initial configuration submission b81c7775-771d-4413-9288-bfcb6840f041 was successful (context: {participantId=sandbox-participant})
2020-11-18T09:50:23.585Z INFO  c.d.p.a.s.LedgerConfigProvider@[ForkJoinPool-20-worker-9] -
                Initial ledger configuration lookup did not find any configuration (context: {participantId=sandbox-participant})
2020-11-18T09:50:23.615Z INFO  c.d.p.a.LedgerApiServer@[kka.actor.default-dispatcher-7] -
                Listening on localhost:51013 over plain text. (context: {participantId=sandbox-participant})
   ____             ____
  / __/__ ____  ___/ / /  ___ __ __
 _\ \/ _ `/ _ \/ _  / _ \/ _ \\ \ /
/___/\_,_/_//_/\_,_/_.__/\___/_\_\

2020-11-18T09:50:23.616Z INFO  c.d.p.s.SandboxServer@[kka.actor.default-dispatcher-7] -
                Initialized Sandbox version 0.0.0 with ledger-id = sandbox-server, port = 51013, dar file = List(/private/var/tmp/_bazel_vsts/38d36a619150b9300aa37bd0849c0ccc/sandbox/processwrapper-sandbox/664/execroot/com_github_digital_asset_daml/bazel-out/darwin-opt/bin/ledger/sandbox-classic/sandbox-classic-tests_test_suite_src_test_suite_scala_platform_sandbox_ScenarioLoadingITPostgres.scala.runfiles/com_github_digital_asset_daml/ledger/test-common/model-tests.dar), time mode = static time, ledger = postgres, auth-service = AuthServiceWildcard$, contract ids seeding = no
2020-11-18T09:50:23.617Z WARN  c.d.p.s.SandboxServer@[kka.actor.default-dispatcher-7] -
                Initializing a ledger with scenarios is deprecated and will be removed in the future. You are advised to use DAML Script instead. Using scenarios in DAML Studio will continue to work as expected.
A migration guide for converting your scenarios to DAML Script is available at https://docs.daml.com/daml-script/#using-daml-script-for-ledger-initialization
2020-11-18T09:50:23.697Z INFO  c.d.p.a.s.LedgerConfigProvider@[kka.actor.default-dispatcher-9] -
                New ledger configuration Configuration(1,TimeModel(PT0S,PT30S,PT30S),PT24H) found at Absolute(0000000000000009) (context: {participantId=sandbox-participant})
  - should event ids from the active contracts service can be used to load transactions *** FAILED *** (102 milliseconds)
    io.grpc.StatusRuntimeException: UNAVAILABLE: The ledger configuration is not available.
```

[Full logs.](https://dev.azure.com/digitalasset/daml/_build/results?buildId=61194&view=logs&jobId=a5e52b91-c83f-5429-4a68-c246fc63a4f7&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=5add3200-7efa-59c0-0a45-43bce0c77c69)

So if the server isn't up until `LedgerConfigProvider#ready` starts, then I have to go back to my original guess: that there's a race condition there.

This issue is in Sandbox Classic, where the config timeout is set to 5 seconds. Here the test fails in hundreds of milliseconds, so I don't think that's it.